### PR TITLE
Adds zeppelin tests

### PR DESCRIPTION
**Pull request by joshuata**

### CODE REVIEW

File: magpie/setup/magpie-setup-project-spark [hunk 1]
```diff
@@ -487,7 +487,7 @@ fi
 
 if [ "${ZEPPELIN_SETUP}" == "yes" ]
 then
-    zeppelin_jars="${ZEPPELIN_HOME}/zeppelin-server-${ZEPPELIN_VERSION}-incubating.jar"
+    zeppelin_jars="${ZEPPELIN_HOME}/zeppelin-server-${ZEPPELIN_VERSION}.jar"
     zeppelin_jars="${zeppelin_jars}:${ZEPPELIN_HOME}/lib/*"
     if [ "${sparkclasspath}X" != "X" ]
     then
```
Comment (chu11) [inline magpie/setup/magpie-setup-project-spark]: could we split this into a separate patch, since it's clearly a bug fix separate from the testsuite additions.  Could be a separate pull request if you want.

File: testsuite/test-generate-zeppelin.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    cp ../submission-scripts/script-${submissiontype}/magpie.${submissiontype}-spark-with-zeppelin magpie.${submissiontype}-spark-with-zeppelin-${zeppelinversion}-${sparkversion}
+
+    sed -i -e 's|export MAGPIE_JOB_TYPE="\(.*\)"|export MAGPIE_JOB_TYPE="script"|' magpie.${submissiontype}-spark-with-zeppelin-${zeppelinversion}-${sparkversion}
+
+    sed -i -e 's|export SPARK_VERSION="\(.*\)"|export SPARK_VERSION="'"${sparkversion}"'"|' magpie.${submissiontype}-spark-with-zeppelin-${zeppelinversion}-${sparkversion}
+
+    sed -i -e 's|export SPARK_MODE="\(.*\)"|export SPARK_MODE="script"|' magpie.${submissiontype}-spark-with-zeppelin-${zeppelinversion}-${sparkversion}
+
+    sed -i -e 's|# export SPARK_SCRIPT_PATH="\(.*\)"|export SPARK_SCRIPT_PATH="testscripts/test-zeppelin.py"|' magpie.${submissiontype}-spark-with-zeppelin-${zeppelinversion}-${sparkversion}
+
```
Comment (chu11) [inline testsuite/test-generate-zeppelin.sh]: shouldn't this be "MAGPIE_SCRIPT_PATH" since we're using MAGPIE_JOB_TYPE=script?

File: testsuite/test-generate-zeppelin.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    sed -i -e 's|export ZEPPELIN_VERSION="\(.*\)"|export ZEPPELIN_VERSION="'"${zeppelinversion}"'"|' magpie.${submissiontype}-spark-with-zeppelin-${zeppelinversion}-${sparkversion}
+
+    sed -i -e 's|export ZEPPELIN_HOME="\(.*\)"|export ZEPPELIN_HOME="'"${zeppelinhome}"'"|' magpie.${submissiontype}-spark-with-zeppelin-${zeppelinversion}-${sparkversion}
+
```
Comment (chu11) [inline testsuite/test-generate-zeppelin.sh]: I don't think this needs to be set anymore now that the ZEPPELIN_VERSION is set correctly?  Am I wrong?

File: magpie/setup/magpie-setup-project-spark (hunk 1 above)
Comment (joshuata) [inline magpie/setup/magpie-setup-project-spark]: Okay. I might be easier for you to do it then, since I have to do pull requests for everything.

File: testsuite/test-generate-zeppelin.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+source test-config.sh
+
+__GenerateZeppelinStandardTests_BasicTests() {
+    local sparkversion=$1
+    local javaversion=$2
+    local zeppelinversion=$3
+    local zeppelinhome=$4
+
+    cp ../submission-scripts/script-${submissiontype}/magpie.${submissiontype}-spark-with-zeppelin magpie.${submissiontype}-spark-with-zeppelin-${zeppelinversion}-${sparkversion}
+
```
Comment (chu11) [inline testsuite/test-generate-zeppelin.sh]: For consistency to other generated files, could we do:

`magpie.${submissiontype}-spark-with-zeppelin-${zeppelinversion}-spark-${sparkversion}`

so the second version number is clearly identified as a spark version.

File: testsuite/testscripts/test-zeppelin.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env python
+
+import os
+import urllib2
+
+master_url = "http://" + os.environ['ZEPPELIN_MASTER_NODE'] + ":" + os.environ['default_zeppelin_port']
+
```
Comment (chu11) [inline testsuite/testscripts/test-zeppelin.py]: probably want to use ZEPPELIN_SERVER_PORT now?

File: testsuite/testscripts/test-zeppelin.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+response = urllib2.urlopen(master_url).getcode()
+
+if response == 200:
+    return 0
```
Comment (chu11) [inline testsuite/testscripts/test-zeppelin.py]: hmmm, I think we'll need a print message indicating success?  So that the test can be validated in test-validate.sh later on?

File: magpie/setup/magpie-setup-project-spark (hunk 1 above)
Comment (chu11) [inline magpie/setup/magpie-setup-project-spark]: ok, i'll fix.

File: testsuite/testscripts/test-zeppelin.py (hunk 6 above)
Comment (joshuata) [inline testsuite/testscripts/test-zeppelin.py]: Done

File: testsuite/testscripts/test-zeppelin.py (hunk 5 above)
Comment (joshuata) [inline testsuite/testscripts/test-zeppelin.py]: Done

File: testsuite/test-generate-zeppelin.sh (hunk 4 above)
Comment (joshuata) [inline testsuite/test-generate-zeppelin.sh]: Done

File: testsuite/test-generate-zeppelin.sh (hunk 3 above)
Comment (joshuata) [inline testsuite/test-generate-zeppelin.sh]: Correct. And completed.

File: testsuite/test-generate-zeppelin.sh (hunk 2 above)
Comment (joshuata) [inline testsuite/test-generate-zeppelin.sh]: Fixed

File: magpie/setup/magpie-setup-project-spark (hunk 1 above)
Comment (chu11) [inline magpie/setup/magpie-setup-project-spark]: pushed upstream

Comment (joshuata) [inline magpie/setup/magpie-setup-project-spark]: Merged 😄

File: testsuite/testscripts/test-zeppelin.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import os
+import urllib2
+
+master_url = "http://" + os.environ['ZEPPELIN_MASTER_NODE'] + ":" + os.environ['ZEPPELIN_SERVER_PORT']
+
+response = urllib2.urlopen(master_url).getcode()
+
+if response == 200:
+    print "Success"
+    return 0
```
Comment (chu11) [inline testsuite/testscripts/test-zeppelin.py]: See my upcoming comment on test-validation, but perhaps these print statements will need to be enhanced.

**Comment by chu11** [comment]
Overall looks good, just a few things to finish up.

1) So we need to add test-validation of the zeppelin outputs into `test-validate.sh`.  You can probably copy something very similar to what is used by zookeeper in that file to get an idea of what to do.

2) You'll probably need to put something into the post processing to give the test a suitable filename to get searched for.  `run-zeppelinurlopen` seems like a reasonable thing to try.

3) could you update README.md and doc/README to indicate 0.6.0 Zeppelin is now a version supported and update the git commit message appropriately to indicate this new support.
